File: graphql/graphql08_kickstarter_mutation/src/main/java/com/sc/graphql/service/EmployeeService.java
package com.sc.graphql.service;

import java.util.List;
import java.util.Optional;

import com.sc.graphql.entity.Employee;
import com.sc.graphql.entity.EmployeeInput;
import com.sc.graphql.repository.EmployeeRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class EmployeeService {
    private final EmployeeRepository employeeRepository;

    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }

    public List<Employee> findByDepartmentId(Long departmentId) {
        return employeeRepository.findByDepartmentId(departmentId);
    }

    public Employee addEmployee(EmployeeInput employeeInput) {
        Employee employee = new Employee(null,
                employeeInput.getName(),
                employeeInput.getSalary(),
                employeeInput.getDepartmentId());

        return employeeRepository.save(employee);
    }

    public Employee deleteEmployee(Long id) {
        return employeeRepository.findById(id)
                .map(e -> {
                    employeeRepository.deleteById(e.getId());
                    return e;
                }).orElse(null);
    }

    public Employee updateEmployee(Long id, EmployeeInput employeeInput) {
        Employee employee = new Employee(id, employeeInput.getName(), employeeInput.getSalary(), employeeInput.getDepartmentId());
        return employeeRepository.save(employee);
    }
}

File: kotlin/kotlin_springboot01/src/main/kotlin/com/sc/kotlin/service/EmployeeService.kt
package com.sc.kotlin.service

import com.sc.kotlin.entity.Employee
import com.sc.kotlin.repository.EmployeeRepository
import org.springframework.stereotype.Service

interface EmployeeService {
    fun findAll(): List<Employee>
    fun save(employee: Employee): Employee
}

@Service
class EmployeeServiceImpl(
        private val employeeRepository: EmployeeRepository
) : EmployeeService {

    override fun findAll(): List<Employee> {
        return employeeRepository.findAll()
    }

    override fun save(employee: Employee): Employee {
        return employeeRepository.save(employee)
    }
}
